src/elements/button/button.ts

```diff
@@ -0,0 +1,76 @@
+///<reference path="../../../typings/angularjs/angular.d.ts"/>
+///<reference path="../../../typings/angularjs/angular-animate.d.ts"/>
+
+'use strict';
+
+class SmButtonController {
+  static $inject = ['$animate'];
+
+  constructor(public $animate: ng.IAnimateCallbackObject) { }
```

Comment:
I just have a quick look. It's working well to me.

I have some questions:
- Should we use tsconfig.json for compiler options? It would be good for IDE support . I haven't looked into webpack support, but I think it should be possible.
- Is there a way to start the demo?